mne_bids/read.py

```diff
@@ -151,9 +151,20 @@ def _read_events(events, event_id, raw, bids_path=None):
                 'To specify custom event codes, please pass "event_id".'
             )
         else:
+            special_annots = {"BAD_ACQ_SKIP"}
             desc_without_id = sorted(
                 set(raw.annotations.description) - set(event_id.keys())
             )
+            # auto-add entries to `event_id` for "special" annotation values
+            # (but only if they're needed)
+            if set(desc_without_id) & special_annots:
+                existing_ids = set(event_id.values())
+                for annot in special_annots:
+                    # don't change value if key is present; use a value guaranteed to
+                    # not be in use
+                    event_id.setdefault(annot, max(existing_ids) + 1)
```

Comment:
I'm wondering now about this approach. this will modify the user-passed dictionary. Stand by while I push a safer alternative.